File: frontend/src/objects/user/Customer.ts
import type { OrderId, UserId } from '@/objects/shared/ids'
import type { Voucher } from '../shared/Voucher'

export interface Customer {
  id: UserId
  name: string
  phone: string
  defaultAddress: string
  walletBalance: number
  orderHistoryIds: OrderId[]
  vouchers: Voucher[]
}

File: backend/src/user/objects/Customer.scala
package delivery.user.objects

import delivery.shared.objects.{OrderId, UserId, Voucher}

final case class Customer(
    id: UserId,
    name: String,
    phone: String,
    defaultAddress: String,
    walletBalance: Double,
    orderHistoryIds: List[OrderId],
    vouchers: List[Voucher]
)
